### DISCUSSION TITLE
Converting msdf generator atlas storage to texture in memory.

### DISCUSSION THREAD

**Discussion by Stanlyhalo** [Chlumsky/msdf-atlas-gen#57] [Q&A] [answered]
This isn't exactly an issue with msdf-atlas-gen itself, so I'm posting in discussions. Essentially what I'm trying to do, is take the atlas storage, and use the pixels to convert it into a texture. Here's the line of code to create the generator:
![image](https://user-images.githubusercontent.com/36088749/204614005-9488dea7-527a-42c7-bc7b-b40599819a14.png)

And after a bit, I attempt to use the atlasStorage to create a vector of uint32_t. For reference, if I set all of the uint32_t entries to `0xffffffff`, it's all white, which is what I want. But here's my code for converting the unsigned char (byte) to uint32_t (if I just set the index in the data variable to ref.pixels[i], it seems sorta correct, but it's all red):
![image](https://user-images.githubusercontent.com/36088749/204614325-34e88c64-38fc-445f-a652-b000ddc206cc.png)

Here's my example of what it should look like via blender:
![image](https://user-images.githubusercontent.com/36088749/204614463-e9575539-61c0-43cc-9fbd-f46dadea2ae4.png)

And here's the result I get in the viewport:
![image](https://user-images.githubusercontent.com/36088749/204614511-ac7a7358-e021-4d0e-9033-cc6569ae24b9.png)

Maybe I'm not understanding how the pixels are sorted in the array, but this is the result, and I can't find any solutions online, and I have been getting help from several other people, but they too are baffled by the issues. Again, my take is it's just the arrangement of the pixels in some weird way.

**Comment by Chlumsky** [comment, marked as answer]
`ref.pixels[4*i], ref.pixels[4*i+1], ...`

**Reply by Stanlyhalo** [reply]
Okay, wow, I just missed those two character in each line. I'm embarrassed lol. Thanks for the help anyways hehe
